openmsupply/application-manager-web-app
# 386 update use get review structure to generate progress and do validation for lvl 1 review (first review)

Fixes #386 
Branched From: #385 

Requires back end #386frontend-helpers for testing and maybe that branch should be PRed as it's quite helpful (couldn't find any other application/reviews that haven't been started)

### Changes

* getReview query now return question assignments
* generateReviewProgress helpers
* useGetFullReview structure to use helpers and to add isAssigned to element (something new we haven't discussed but is super useful, also we should make sure to add question assignments when self assigning

### Testing

* use test branch (386 test) and back end branch
* navigate to http://localhost:3000/applicationNew/ABC123/reviewTest
* in ReviewPageWrapper test use review assignment 1020 (reviewer 1) or 1021 (reviewer 2), can change those two and logged in users to see initial status as all 0
* can start review (previous way) http://localhost:3000/application/ABC123/review, with either reviewer 1 or reviewer 2
* progress should change yet
* can add reviews and see progress, firstIncomplteReviewPage, and canSubmitReviewAs change (the later one will only change if can be submitted, otherwise it's null/undefined)

### Notes

Progress is calculated from application response (latest) having a reviewResponse (latest and submitted or draft belonging to current user) that has a decision. So this would work for consolidator as-well (i.e. can see on home page, review progress /decision for submitted lvl 1 reviewers).
